argoverse/utils/transform.py

```diff
@@ -26,6 +26,23 @@ def quat2rotmat(q: np.ndarray) -> np.ndarray:
         R: Array of shape (3, 3) representing a rotation matrix.
     """
     assert np.isclose(np.linalg.norm(q), 1.0, atol=1e-12)
+
+    quat_xyzw = quat_argo2scipy(q)
+    return Rotation.from_quat(quat_xyzw).as_matrix()
+
+
+def quat_argo2scipy(q: np.ndarray) -> np.ndarray:
+    """Re-order a scalar-first [w, x, y, z] quaternion format used by Argoverse to the scalar-last format in SciPy."""
     w, x, y, z = q
     q_scipy = np.array([x, y, z, w])
-    return Rotation.from_quat(q_scipy).as_dcm()
+    return q_scipy
+
+
+def quat_argo2scipy_vectorized(q: np.ndarray) -> np.ndarray:
+    """Re-order a scalar-first [w, x, y, z] quaternion format used by Argoverse to the scalar-last format in SciPy."""
+    return q[..., [1, 2, 3, 0]]
+
+
+def quat_scipy2argo_vectorized(q: np.ndarray) -> np.ndarray:
+    """Re-order a scalar-first [w, x, y, z] quaternion format used by Argoverse to the scalar-last format in SciPy."""
```

Comment:
i dont think the docstrings are right on these 3 functions. I would write them as follows:

```python
def quat_argo2scipy()
      """Re-order Argoverse's scalar-first [w,x,y,z] quaternion order to Scipy's scalar-last [x,y,z,w]
```

```python
def quat_argo2scipy_vectorized()
      """"Re-order Argoverse's scalar-first [w,x,y,z] quaternion order to Scipy's scalar-last [x,y,z,w]
```

```python
def quat_scipy2argo_vectorized()
     """"Re-order Scipy's scalar-last [x,y,z,w] quaternion order to Argoverse's scalar-first [w,x,y,z].
```